core/unix/signal.c

```diff
@@ -674,9 +674,14 @@ create_clone_record(dcontext_t *dcontext, reg_t *app_thread_xsp, app_pc thread_f
 create_clone_record(dcontext_t *dcontext, reg_t *app_thread_xsp,
                     /* Do not store dr_clone_args; it is freed in the parent thread
                      * in the post-syscall handling of clone3.
+                     *
+                     * These args are not really clone3_syscall_args_t*. The app
+                     * uses the clone_args provided by the kernel, which may have
+                     * more fields than DR's internal struct (clone3_syscall_args_t)
+                     * We also don't want to use the kernel's clone_args struct here,
+                     * because it may not be available on all LINUX versions.
                      */
-                    clone3_syscall_args_t *dr_clone_args,
-                    clone3_syscall_args_t *app_clone_args)
+                    void *dr_clone_args, void *app_clone_args)
```

Comment:
@derekbruening Looking for some feedback on this. Calling these `clone3_syscall_args_t*` doesn't seem correct, though it would make some of the code below look better.

Other DR-internal copies of kernel structs usually are meant to be kept in sync with their kernel version. `clone3_syscall_args_t` doesn't need to be. This is why I thought maybe it shouldn't be an alias for `clone_args`.

Reply:
See above: I would use the type.  It is a standard practice for these sized structs to have varying fields off the end without having to throw away type info and have a formless blob.

Reply:
Done.